crates/libcontainer/src/channel.rs

```diff
@@ -198,18 +201,20 @@ where
 {
     let (os_sender, os_receiver) = unix_channel()?;
     let receiver = Receiver {
-        receiver: os_receiver,
+        receiver: os_receiver.as_raw_fd(),
         phantom: PhantomData,
     };
     let sender = Sender {
-        sender: os_sender,
+        sender: os_sender.as_raw_fd(),
         phantom: PhantomData,
     };
+    std::mem::forget(os_sender);
+    std::mem::forget(os_receiver);
```

Comment:
> I use forget() because some functions in the nix library now return an OwnedFd. This helps avoid accidentally closing the file descriptor too soon by preventing its automatic drop. Maybe fully relying on OwnedFd to manage the fd's lifetime instead of manually calling close on RawFd is a better approach, but I think this could require quite a bit of code changes.

What should we do to support `OwnedFd`? I feel it is too dangerous for low-level container runtime because we must take care of the security risk.

Reply:
For `Sender` and `Receiver`, the problem is relatively easy. I noticed that `Sender::close()` will closes the fd.

I don't have a solid plan for fully supporting `OwnedFd`. I can only say that I've tried to make my PR as compatible as possible with the previous implementation that used `rawfd`, and reuse the code that previously managed file descriptors.

The main difficulty right now is that if we changes entirely to `OwnedFd`, the `console_socket` in `struct ContainerArgs` (crates/libcontainer/src/process/args.rs:20) won't satisfy the `Clone trait`. Maybe need to introduce `Rc<T>`.

Reply:
Should I avoid `forget()` in this PR, or address it in a separate issue? I'd like to hear your opinion.

Reply:
> Should I avoid forget() in this PR, or address it in a separate issue? I'd like to hear your opinion.

Thanks for your detailed investigation. This problem relates to the security issue. So I can't merge it before removing `forget()`.

Reply:
I agree with this cautious decision. I have some ideas for removing `forget()`, but it will take some time to implement.

Reply:
Thanks for understanding 🙏 

> but it will take some time to implement.

NP 👍 Let's give it a try!